src/app/trainees/trainees-routing.module.ts

```diff
@@ -0,0 +1,25 @@
+import { NgModule } from "@angular/core";
+import { RouterModule, Routes } from "@angular/router";
+import { TraineesComponent } from "./trainees.component";
+import { TraineeListComponent } from "./trainee-list/trainee-list.component";
+
+const routes: Routes = [
+  {
+    path: "trainees",
+    component: TraineesComponent,
+    data: { title: "Trainees" },
+    children: [
+      {
+        path: "column/:sortColumn/direction/:sortDirection",
```

Comment:
these have been made mandatory where I feel they are optional i.e. path could be "" with optional {sortcolumn, sortdirection}  which will look like "http://localhost:4200/trainees/;submissionDate;asc" I think we should discuss routing so the app is not disjointed. "http://localhost:4200/trainees/:id" should get trainee with id. now this isn't optional and "http://localhost:4200/trainees" should get all trainees, no filters etc.

Reply:
good point @ojemuyiwa - been doing some reading here https://angular-2-training-book.rangle.io/routing/query_params and it seems that query params are the way to go to make them optional as route param variables behave as essential to determining route

I have made the params optional via query strings which means now `http://localhost:4200/trainees` also gets all trainees with no filters or any other params.

pr updated